### PR TITLE
Added favicons and dark mode for search

### LINKED ISSUE #79: Adding a favicon to the website.
### Is your feature request related to a problem? Please describe

The website currently lacks a favicon, which is the small icon displayed in the browser tab or next to the website name in bookmarks.

### Describe the solution you'd like

Design and Implement a Favicon: Create a favicon.ico file with the desired icon and dimensions (usually 16x16 pixels). This file should be placed in the root directory of the website.

### Describe alternatives you've considered

Update HTML Code: Insert the following code within the <head> section of each HTML page to link to the favicon:

### Additional context

_No response_

### Are you working on this?

Yes

### LINKED ISSUE #85: feat:  In light mode and dark mode, the Algolia search bar is in light mode.
### Is your feature request related to a problem? Please describe

In dark mode, the Algolia search bar should be in dark mode, and in light mode, the Algolia search bar should be in light mode.

Light

Dark

### Describe the solution you'd like

The Algolia search bar also supports dark mode.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by Ajay9330**
#### Description:
all types of favicons added and fixes the previous missing part of #78  dark mode of search

#### Issue fixed:
 - fixes #85 and
 - fixes #79  

#### Changes done: 
- [x] favicons added
- [x] added dark mode for search

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Tried squashing the commits into one

### CODE REVIEW

**Comment by Ajay9330** [comment]
sir @aaronbrethorst  please check this

File: src/_layouts/default.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html lang="<%= site.locale %>"  x-data="{ themeMode: window.localStorage.getItem('mode') || (window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light') }" :class="{'dark':themeMode=='dark','hidden':!themeMode}" class='hidden h-full' >
+<html lang="<%= site.locale %>"  x-data="{ themeMode: window.localStorage.getItem('mode') || (window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light') }" :class="{'dark':themeMode=='dark','hidden':!themeMode}":data-theme=themeMode class='hidden h-full' >
```
Comment (aaronbrethorst) [inline src/_layouts/default.erb:2]: you need to watch your spacing and quoting on changes like this.

**Review by aaronbrethorst — APPROVED** [review]
I'm going to merge this, but there are a few issues with this PR: 

1. you need to watch the way you lay out HTML attributes. Include one space between each attribute. Always quote (single or double) your attribute's values.
2. The favicons have been derived from a low-resolution source and look like they've been upscaled.

I'll fix both of these separately.

**Comment by aaronbrethorst** [comment]
@Ajay9330 also, please _do not_ combine unrelated fixes in a single commit.

File: src/_layouts/default.erb (hunk 1 above)
Comment (Ajay9330) [reply src/_layouts/default.erb:2]: ok sorry sir i done mistake again due to word-wrap on in vs code

**Comment by Ajay9330** [comment]
Actually, i Tried squashing the commits into one as mentioned in the PR rule, but i'll  keep only related fixes in single commit

**Pull request merged**